Transformers/Java/src/test/java/com/github/ciselab/lampion/transformations/RenameVariableTransformerTests.java

```diff
@@ -0,0 +1,276 @@
+    }
+
+    @Test
+    void testExclusive_isExclusiveWithNothing(){
+        RenameVariableTransformer transformer = new RenameVariableTransformer();
+
+        assertTrue(transformer.isExclusiveWith().isEmpty());
+    }
+
+    @Test
+    void constraintsAreNotSatisfied_NoVariablesToChange_ReturnsEmptyResult(){
+        CtClass emptyClass = Launcher.parseClass("package lampion.test.examples; class A { int addOne(int a) { return a + 1 }");
+
+        RenameVariableTransformer transformer = new RenameVariableTransformer();
+        TransformationResult result = transformer.applyAtRandom(emptyClass);
+
+        assertEquals(new EmptyTransformationResult(), result);
+    }
+
+    @Test
+    void applyToMethod_CheckTransformationResult_nameIsRandomVariableName(){
+        CtElement ast = addOneLocalVariableExample();
+
+        RenameVariableTransformer transformer = new RenameVariableTransformer();
+
+        TransformationResult result = transformer.applyAtRandom(ast);
+
+        assertEquals("RenameVariableTransformer",result.getTransformationName());
+    }
+
+    @Test
+    void applyToMethod_CheckTransformationResult_categoriesNotEmpty(){
+        CtElement ast = addOneLocalVariableExample();
+
+        RenameVariableTransformer transformer = new RenameVariableTransformer();
+
+        TransformationResult result = transformer.applyAtRandom(ast);
+
+        assertFalse(result.getCategories().isEmpty());
+    }
+
+    @Test
+    void applyWithoutDebugSettingOn_TransformationResultShouldHaveNoOptionalInfo(){
+        CtElement ast = addOneLocalVariableExample();
+
+        RenameVariableTransformer transformer = new RenameVariableTransformer();
+
+        transformer.setDebug(false);
+
+        TransformationResult result = transformer.applyAtRandom(ast);
+
+        assertFalse(result.getInitialScopeOfTransformation().isPresent());
+        assertFalse(result.getBeforeAfterComparison().isPresent());
+    }
+
+    @Test
+    void applyWithDebugSettingOn_TransformationResultShouldHaveMoreInfo(){
+        CtElement ast = addOneLocalVariableExample();
+
+        RenameVariableTransformer transformer = new RenameVariableTransformer();
+
+        transformer.setDebug(true);
+
+        TransformationResult result = transformer.applyAtRandom(ast);
+
+        assertTrue(result.getInitialScopeOfTransformation().isPresent());
+        assertTrue(result.getBeforeAfterComparison().isPresent());
+    }
+
+    @Test
+    void testGetStringRandomness_onFullRandom_ShouldGiveFullRandom(){
+        RenameVariableTransformer transformer = new RenameVariableTransformer();
+        transformer.setFullRandomStrings(true);
+
+        assertTrue(transformer.isFullRandomStrings());
+    }
+
+    @Test
+    void testGetStringRandomness_onPseudoRandom_ShouldGivePseudoRandom(){
+        RenameVariableTransformer transformer = new RenameVariableTransformer();
+        transformer.setFullRandomStrings(false);
+
+        assertFalse(transformer.isFullRandomStrings());
+    }
+
```

Comment:
One Test that is missing: 
Does the method only change once after renaming the variable? 
(The test that variables are saved in the transformer and checked by the constraints)